ssh/tailssh/tailssh.go

```diff
@@ -1030,6 +1031,12 @@ func (ss *sshSession) handleSSHAgentForwarding(s ssh.Session, lu *userMeta) erro
 	if err != nil {
 		return err
 	}
+	if uid > math.MaxInt {
```

Comment:
Since we're parsing them with `ParseUint` and passing 32 for the width, use `math.MaxUint32`.

However, we could also parse them as 64-bit unsigned, which some systems do use.

Reply:
ParseInt docs:

> Bit sizes 0, 8, 16, 32, and 64 correspond to int, int8, int16, int32, and int64.

Change the 32 above to 0?

Reply:
On linux, UID/GID are 32-bit uints.
However, we have to pass them to `os.Chown` below which takes `int`. On 32-bit systems we lose 1 bit because of the sign.
We can't really change the signature of `os.Chown` to take a `uint`, unfortunately.

Reply:
Ah. That's worth a comment then.

Reply:
Sure, added a comment.

Reply:
Might be worth logging ("cannot represent actual UID to Go"). On the extremely unlikely chance the bug ever happens, it'll be hard to diagnose without a clue in the logs.

OR, maybe it's fine, because `int` is 64 bits on 64-bit machines, and maybe Go just passes the word to the OS as-is, and the kernel casts signed to unsigned losslessly, and we're actually all good to `ParseUint64`?

Reply:
Is it only 4294967295 that we can't handle?

```
dev:~ $ cat chown.go
package main

import (
	"fmt"
	"math"
	"os"
)

func main() {
	v := int(math.MaxUint32)
	fmt.Println(v)
	os.Chown("foo.txt", v, v)

	v = int(math.MaxUint32 - 1)
	fmt.Println(v)
	os.Chown("foo.txt", v, v)
}
dev:~ $ go build chown.go 
dev:~ $ strace -e fchownat ./chown 
--- SIGURG {si_signo=SIGURG, si_code=SI_TKILL, si_pid=27110, si_uid=1000} ---
--- SIGURG {si_signo=SIGURG, si_code=SI_TKILL, si_pid=27110, si_uid=1000} ---
4294967295
fchownat(AT_FDCWD, "foo.txt", -1, -1, 0) = -1 ENOENT (No such file or directory)
4294967294
fchownat(AT_FDCWD, "foo.txt", 4294967294, 4294967294, 0) = -1 ENOENT (No such file or directory)
+++ exited with 0 +++
```

Reply:
Or that might just be a strace stringification bug. Maybe there's no problem.

Reply:
@bradfitz is that on a 32 or 64-bit machine?

Reply:
I'm not sure there's even a problem on 32-bit?

```
dev:~ $ ls -l foo.txt
-rw-r--r-- 1 4294967294 4294967294 0 Dec 22 12:20 foo.txt
dev:~ $ cat chown.go
package main

import (
	"fmt"
	"os"
	"syscall"
)

func main() {
	fi, _ := os.Stat("foo.txt")
	sys := fi.Sys().(*syscall.Stat_t)

	fmt.Println(sys.Uid)
	fmt.Println(int(sys.Uid))
	fmt.Println(os.Chown("foo.txt", int(sys.Uid), int(sys.Gid)))

	fi, _ = os.Stat("foo.txt")
	sys = fi.Sys().(*syscall.Stat_t)
	fmt.Println(sys.Uid)
}
dev:~ $ GOARCH=386 go build chown.go
dev:~ $ sudo ./chown 
4294967294
-2
<nil>
4294967294
```

Reply:
Then I think @noncombatant assumption about Go passing the word to the syscall directly is correct.
Seems like this is all safe and we can dismiss https://github.com/tailscale/tailscale/security/code-scanning/121 as false positive?

Reply:
With a comment as to why/link to this PR, yeah. Thanks!